### PR TITLE
Change release documentation and small fix to switcher script

### PR DESCRIPTION
**Pull request by rloewe**
When reviewing this, you should also check the whole `create_switcher.py` script. The commit is not part of this pull request.

### CODE REVIEW

**Review by Chroxvi — CHANGES_REQUESTED** [review]
Generally this looks very good. I have a few suggested changes as well as some bullets for discussion.

First of all, as part of this PR, we should also update https://cotainr.readthedocs.io/en/latest/development/test_suite_ci_cd.html#read-the-docs-continuous-documentation to mention the new RTD automation rule ("CalVer releases") that automatically activates all git tagged versions on RTD.

A few other things we might consider to implement in this PR as well (or in relation to):
- Switch the default version on RTD to "stable".
- Update the release notes (https://cotainr.readthedocs.io/en/latest/release_notes/index.html) to include a linked to the specific version of the docs on RTD, e.g the headline "2023.02.0" could be a link to https://cotainr.readthedocs.io/en/2023.02.0/. If make this change, the `release_note.md.template` should be updated accordingly.
- Changing the version number being shown on the https://cotainr.readthedocs.io/en/latest/index.html to something along the lines of "Unreleased development version (<git_hash>)", i.e. changing the `version` variable in `doc/conf.py` when on  RTD and building the "latest" version. This should be possible by exploiting the RTD environment variables (https://docs.readthedocs.io/en/stable/reference/environment-variables.html)
- Adding a (somewhat unrelated) note to the release guide about updating the `README.md` Easybuild easyconfig example to use the new version of cotainr.

File: doc/create_switcher.py [hunk 1]
```diff
@@ -19,13 +25,13 @@
...
     {
-        "name": f"{tags[0]} (stable)",
+        "name": f"{cotainr.__version__} (stable)",
         "version": "stable",
         "url": "https://cotainr.readthedocs.io/en/stable/",
     },
 ]
 
-for tag in tags[1:4]:
+for tag in tags[0:3]:
```
[the remark this answers is not in the document]
Comment (rloewe) [reply doc/create_switcher.py:34]: It is in the releasing.rst guide. But would you also want a comment in the code?

[the remark this answers is not in the document]
Comment (Chroxvi) [reply doc/create_switcher.py:34]: Yes, please - possibly with a reference to the documentation

**Comment by rloewe** [comment]
I have fixed the issued mentioned in the code. I have also:

- Changed the default documentation to stable
- Added links to release specific documentation in release notes and templates
- Added a note about update README.md EasyBuild script in releasing documentation

I have not done anything about the latest tag. I felt that would be too time-consuming to fix in a good way.

**Review by Chroxvi — APPROVED** [review]
Excellent! I think this is ready to be merged.

Let's leave any changes to the handling of the "latest" tag for another PR.

**Pull request merged**